app/models/upload.rb

```diff
@@ -100,19 +154,22 @@ def extract_contents!(mimetype)
     extracted_path.mkpath
     if mimetype == "application/zip" ||
        submission_path.to_s.ends_with?(".zip")
-      ZipRuby::Archive.open(submission_path.to_s) do |ar|
-        ar.each do |zf|
-          if zf.directory?
-            FileUtils.mkdir_p(extracted_path.join(zf.name))
+      output, err, status = Open3.capture3("unzip", "-n", # Never overwrite
```

Comment:
It might be worth pulling all this archive logic out into an archive handling module in /lib